devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,532 @@
+
+	refreshCommands
+}
+
+def configure() {
+	log.debug "Configuration"
+
+	configureChild()
+
+	def configurationCommands = []
+	// todo: check if following binding is necessary here
+	configurationCommands += "zdo bind 0x${device.deviceNetworkId} 1 0xA 0x201 {${device.zigbeeId}} {}"
+	configurationCommands += "zdo bind 0x${device.deviceNetworkId} 1 0xA 0x405 {${device.zigbeeId}} {}"
+
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, 1, 60, 1) //formerly THERMOSTAT MODE: 0x001C
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE, DataType.INT16, 10, 60, 10)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, COOLING_SETPOINT, DataType.INT16, 1, 60, 10)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, HEATING_SETPOINT, DataType.INT16, 1, 60, 10)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, COOLING_SETPOINT_UNOCCUPIED, DataType.INT16, 1, 60, 10)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, HEATING_SETPOINT_UNOCCUPIED, DataType.INT16, 1, 60, 10)
+	//configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, 0x0A58, 0x10, 1, 300, 1) //GFan
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE, DataType.ENUM8, 1, 60, 1)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_FAN_SPEED, DataType.ENUM8, 1, 60, 1)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE, DataType.ENUM8, 1, 60, 1)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, OCCUPANCY, DataType.ENUM8, 1, 60, null)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_OCCUPANCY, DataType.ENUM8, 1, 60, null)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_EFFECTIVE_OCCUPANCY, DataType.ENUM8, 1, 60, null)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_UI_CONFIGURATION_CLUSTER, TEMPERATURE_DISPLAY_MODE, DataType.ENUM8, 1, 60, 1)
+	configurationCommands += zigbee.configureReporting(THERMOSTAT_CLUSTER, CUSTOM_HUMIDITY, DataType.UINT16, 1, 60, 10)
+	configurationCommands += zigbee.configureReporting(RELATIVE_HUMIDITY_CLUSTER, RELATIVE_HUMIDITY_MEASURED_VALUE, DataType.UINT16, 1, 60, 5)
+
+	delayBetween(getRefreshCommands()+configurationCommands)
+}
+
+def configureChild() {
+	if (!childDevices) {
+		createChildThermostat()
+	}
+}
+
+def getCoolingSetpointRange() {
+	(getTemperatureScale() == "C") ? [12, 37.5] : [54, 100]
+}
+def getHeatingSetpointRange() {
+	(getTemperatureScale() == "C") ? [4.5, 32] : [40, 90]
+}
+
+def getTemperature(value) {
+	if (value != null) {
+		def celsius = Integer.parseInt(value, 16) / 100
+		if (temperatureScale == "C") {
+			celsius//Math.round(celsius)
+		} else {
+			celsiusToFahrenheit(celsius)//Math.round(celsiusToFahrenheit(celsius))
```

Comment:
I know in some other device handlers there is rounding logic. Are we not wanting to do that here? If not, let's remove the comments.

Reply:
@dkirker  Sure. 
Actually setpoints can be changed manually with a step = 0.5
Additionally - since the device displays his values on the screen with only one digit after the decimal point and reports its values with 2 digits after decimal point - it' really looks bad if You want to change setpoints on the ST UI (step = 0.5), for example: 18.72 <-19.22 -> 19.72 
so those values will be rounded to the nearest half (for example: 19.22 >> 19.00; 19.32 >> 19.50)
We will also configure  setpoints initially to rounded default values so the device would work on "rounded" numbers from the very beginning.